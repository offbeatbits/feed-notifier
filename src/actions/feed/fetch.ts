import { Feed } from '../../models';

export const fetch = async (url: string) => {
  const feed = await new Feed(url).fetch();
  return feed;
};
