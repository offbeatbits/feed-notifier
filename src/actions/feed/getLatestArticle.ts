export const getLatestArticle = () => {
  console.log('getLatestArticle'); // eslint-disable-line
};
