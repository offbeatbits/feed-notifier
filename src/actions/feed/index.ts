export * from './fetch';
export * from './getLatestArticle';
