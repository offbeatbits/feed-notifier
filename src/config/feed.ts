export const feedUrl = process.env.FN_FEED_URL!;
