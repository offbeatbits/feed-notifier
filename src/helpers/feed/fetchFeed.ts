import { Feed } from "../../models";

export const fetchFeed = (url: string) => new Feed(url).fetch();
