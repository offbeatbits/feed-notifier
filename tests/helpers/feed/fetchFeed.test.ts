import { fetchFeed } from '../../../src/helpers/feed/fetchFeed';

jest.mock('rss-parser', () => jest.fn().mockImplementation(() => ({
  parseURL: () => Promise.resolve({
    content: {
      lastBuildDate: 'mocked_last_build_date',
    },
    lastBuildDate: 'mocked_last_build_date',
    items: [
      {
        pubDate: 'mocked_last_item_pub_date',
        title: 'mocked_last_item_title',
        link: 'mocked_last_item_link',
      },
    ],
  }),
})));

describe('fetchFeed', () => {
  it('should fetch feed', async () => {
    expect.assertions(1);
    await expect(fetchFeed('sample_url')).toMatchSnapshot();
  });
});
