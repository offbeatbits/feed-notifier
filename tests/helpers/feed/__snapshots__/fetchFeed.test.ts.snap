// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetchFeed should fetch feed 1`] = `
Feed {
  "content": Object {
    "content": Object {
      "lastBuildDate": "mocked_last_build_date",
    },
    "items": Array [
      Object {
        "link": "mocked_last_item_link",
        "pubDate": "mocked_last_item_pub_date",
        "title": "mocked_last_item_title",
      },
    ],
    "lastBuildDate": "mocked_last_build_date",
  },
  "url": "sample_url",
}
`;
